components/bt/host/bluedroid/api/include/api/esp_gap_ble_api.h

```diff
@@ -18,46 +19,61 @@ extern "C" {
-#define ESP_LE_AUTH_REQ_SC_MITM_BOND        (ESP_LE_AUTH_REQ_MITM | ESP_LE_AUTH_REQ_SC_ONLY | ESP_LE_AUTH_BOND)   /*!< 1101 SC with MITM and Bonding*/
-typedef uint8_t   esp_ble_auth_req_t;         /*!< combination of the above bit pattern */
-
-#define ESP_BLE_ONLY_ACCEPT_SPECIFIED_AUTH_DISABLE 0    /*!< authentication disable*/
-#define ESP_BLE_ONLY_ACCEPT_SPECIFIED_AUTH_ENABLE  1    /*!< authentication enable*/
+#define ESP_LE_KEY_NONE      0           /*!< No Link Encryption Key is used. */
+#define ESP_LE_KEY_PENC      (1 << 0)    /*!< Pseudo-Random Number Generator (PRNG) Encryption Key. */
+#define ESP_LE_KEY_PID       (1 << 1)    /*!< Identity Resolving Key (IRK) for Privacy. */
+#define ESP_LE_KEY_PCSRK     (1 << 2)    /*!< Privacy-Enhanced Connection Signature Resolving Key (CSRK). */
+#define ESP_LE_KEY_PLK       (1 << 3)    /*!< Long-Term Key (LTK) for Pairing. */
+#define ESP_LE_KEY_LLK       (ESP_LE_KEY_PLK << 4)      /*!< Long-Term Key (LTK) for Legacy. */
+#define ESP_LE_KEY_LENC      (ESP_LE_KEY_PENC << 4)     /*!< Long-Term Key (LTK) for Link Encryption. */
+#define ESP_LE_KEY_LID       (ESP_LE_KEY_PID << 4)      /*!< Link Identity Key (LID). */
+#define ESP_LE_KEY_LCSRK     (ESP_LE_KEY_PCSRK << 4)    /*!< Link Connection Signature Resolving Key (CSRK). */
+
+typedef uint8_t esp_ble_key_type_t;  /*!< Type definition for BLE key types. */
+
 
 #define ESP_BLE_OOB_DISABLE 0    /*!< disable the out of bond*/
 #define ESP_BLE_OOB_ENABLE  1    /*!< enable the out of bond*/
 
-/// relate to BTM_IO_CAP_xxx in stack/btm_api.h
+/**
+ * @brief Relate to BTM_LE_AUTH_xxx in stack/btm_api.h.
+ * These macros are used in defining the BLE authentication requirements in the ESP32 Bluetooth Low Energy (BLE) stack.
+ */
+#define ESP_LE_AUTH_NO_BOND             0x00        /*!< No bonding required. */
+#define ESP_LE_AUTH_BOND                0x01        /*!< Device is bonded with the peer. */
+#define ESP_LE_AUTH_REQ_MITM            (1 << 2)    /*!< Man-in-the-middle attack required. */
+#define ESP_LE_AUTH_REQ_BOND_MITM       (ESP_LE_AUTH_BOND | ESP_LE_AUTH_REQ_MITM) /*!< Bonding with Man-in-the-middle protection. */
+#define ESP_LE_AUTH_REQ_SC_ONLY         (1 << 3)    /*!< Secure connection (SC) only. */
+#define ESP_LE_AUTH_REQ_SC_BOND         (ESP_LE_AUTH_BOND | ESP_LE_AUTH_REQ_SC_ONLY) /*!< Bonding with Secure connection. */
+#define ESP_LE_AUTH_REQ_SC_MITM         (ESP_LE_AUTH_REQ_MITM | ESP_LE_AUTH_REQ_SC_ONLY) /*!< Secure connection with MITM protection. */
+#define ESP_LE_AUTH_REQ_SC_MITM_BOND    (ESP_LE_AUTH_REQ_MITM | ESP_LE_AUTH_REQ_SC_ONLY | ESP_LE_AUTH_BOND) /*!< Secure connection with MITM and Bonding. */
+typedef uint8_t esp_ble_auth_req_t;    /*!< Type definition for bluetooth low energy authentication requirements. */
+
+#define ESP_BLE_ONLY_ACCEPT_SPECIFIED_AUTH_DISABLE    0    /*!< Authentication disabled. */
+#define ESP_BLE_ONLY_ACCEPT_SPECIFIED_AUTH_ENABLE     1    /*!< Authentication enabled. */
```

Comment:
This means only accept specified authentication requirement. If the authentication can't meet with the requirement, then will refuse pairing.